### PR TITLE
feat: add notation plugin uninstall command

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR is based on the spec PR: https://github.com/notaryproject/notation/pull/809

It adds the notation plugin uninstall command.

Tests: The e2e test cases are also included in this PR.

<del>Note: This PR depends on the notation-go PR: https://github.com/notaryproject/notation-go/pull/363

Update: PR https://github.com/notaryproject/notation-go/pull/363 has been merged.

### CODE REVIEW

**Comment by Two-Hearts** [comment]
This PR was reviewed here: https://github.com/notaryproject/notation/pull/832, however, due to some GitHub issue, I accidently deleted the original branch and overwrite it with the current one (the contents of these two PRs are the same).

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/notation/plugin/uninstall.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return command
+}
+
+func uninstallPlugin(command *cobra.Command, opts *pluginUninstallOpts) error {
+	// set logger
+	ctx := opts.LoggingFlagOpts.InitializeLogger(command.Context())
+	pluginName := opts.pluginName
+	exist, err := checkPluginExistence(ctx, pluginName)
+	if err != nil {
+		return fmt.Errorf("failed to check plugin existence: %w", err)
```
Comment (priteshbandi) [inline cmd/notation/plugin/uninstall.go:74]:
```suggestion
		return fmt.Errorf("failed to check plugin existence: %w", err)
```

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**